protocol-units/bridge/shared/src/types.rs

```diff
@@ -135,13 +138,76 @@ impl From<Uint<256, 4>> for TimeLock {
 }
 
 #[derive(Deref, DerefMut, Debug, Clone, Copy, PartialEq, Eq)]
-pub struct Amount(pub u64);
+pub struct Amount(pub AssetType);
+#[derive(Clone, Debug, PartialEq, Eq, Copy)]
```

Comment:
Let's add doc comments for AssetType and its fields
```suggestion
/// The type of Asset being used
#[derive(Clone, Debug, PartialEq, Eq, Copy)]
```